### PR TITLE
Change to build on amazon-linux

### LINKED ISSUE #5: [PROXY] Build on amazon linux
ubuntu's squid package is built without https proxy support (as opposed to https connect proxy). Amazon linux (RHEL) is build with https proxy support, although the package version is much older. To get https proxy support (required by perl AKA Bugzilla) the proxies can be rebuilt on amazon linux.

The underlying puppet module will need to be rebased as there are config settings that are necessary for building on an older version of squid.

### PR DESCRIPTION
**Pull request by tinnightcap**
Fixes #5

### CODE REVIEW

File: nubis/puppet/00_params.pp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+case $::osfamily {
+    'RedHat': {
+        $service_name                   = 'squid'
+        $use_deprecated_opts            = true
+        $makepasswd_package_version     = '5.44.1.15-5.12.amzn1'
+        $makepasswd_package_name        = 'expect'
```
Comment (limed) [inline nubis/puppet/00_params.pp]: Using expect as a makepasswd package? Correct me if I'm wrong but thats the programming language

Comment (tinnightcap) [inline nubis/puppet/00_params.pp]: Well it is a script utility that is used for testing languages, I would not call it a language in of itself. If you know of a better or cleaner way of getting mkpasswd in a RH variant I will gladly incorporate it. This was the best solution I found in my research, but not being a RH fanboy i might have missed something.

Comment (limed) [inline nubis/puppet/00_params.pp]: Well many ways to skin a cat in this instance:

```bash
cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w ${1:-16} | head -n 1
```

```bash
openssl rand -base64 32
```

Or even using pwgen which I'm sure both OS has these packages. However its not a big enough deal to block this merge.

File: nubis/puppet/01_proxy.pp [hunk 2]
```diff
@@ -1,17 +1,20 @@
+# Configuring general stuff
 
-
-exec { 'apt-get update':
-  command => '/usr/bin/apt-get update',
+exec { 'package_manager_update':
+  command => $package_manager_update_command,
```
Comment (limed) [inline nubis/puppet/01_proxy.pp]: Assuming puppet knows how to grab the `params.pp` file here?

Comment (tinnightcap) [inline nubis/puppet/01_proxy.pp]: Yes, well actually I cheat a bit here. I wanted to use the params.pp parlance, however it does not really seem supported when using the directory loading method. No biggie, I just named the params file 00-params.pp to ensure it is evaluated prior to everything else. So in that way, yea it knows how to grab the params.pp file :)

**Comment by limed** [comment]
r+

**Pull request merged**